src/Illuminate/Http/FileHelpers.php

```diff
@@ -47,7 +47,7 @@ public function hashName($path = null)
 
         $hash = $this->hashName ?: $this->hashName = Str::random(40);
 
-        if ($extension = $this->guessExtension()) {
+        if ($extension = $this->guessExtension() ?? $this->getClientOriginalExtension()) {
```

Comment:
The symfony docs warn that `getClientOriginalExtension` is not safe and is user controlled, so should not be used.

Reply:
https://github.com/symfony/symfony/blob/v4.4.8/src/Symfony/Component/HttpFoundation/File/UploadedFile.php#L93-L100

Reply:
Yes, but when we want to save file we dont need to check extension for safety. 

When it really need - check it manually.

Currently i see unexpected behavior - some files stored with extension, some - not.

Reply:
Another way to solve this - allow to pass extension in options to use for saving.

Reply:
We maybe could check if it matches a regex before using it, to only use cases we don't think have been messed with. Like, maybe something like "between 1 and 5 characters in length, and only a-z1-9 (no whitespace or anything else)".